File: 374_GuessNumberHigherOrLower.go
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * func guess(num int) int;
 */

func guessNumber(n int) int {
    low := 1
    mid := (n + low) / 2

    for guess(mid) != 0 {
        if guess(mid) < 0 {
            n = mid - 1
        } else {
            low = mid + 1
        }
        mid = (n + low) / 2
    }
    return mid
}

File: 374_GuessNumberHigherorLower.c
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */
 int guess(int num);
int guessNumber(int n){
	int left = 1;
    int right = n;
    while(left <= right){
        int mid = left + (right - left) / 2;
        if(guess(mid) == -1){
            right = mid - 1;
        } else if (guess(mid)){
            left = mid + 1;
        } else {
            return mid;
        }
    }
    return left;
}
